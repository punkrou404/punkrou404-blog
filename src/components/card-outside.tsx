import React from 'react';
import { ReactNode } from 'react';

type P = {
    children?: ReactNode;
    action?: boolean;
};

export const CardOutside = ({ children, action = true }: P): JSX.Element => {
    const style = action ? 'hover:bg-gray-200 active:bg-gray-400' : '';

    return (
        <div className="w-full flex-col p-3 text-gray-500">
            <div className={`p-4 bg-white rounded-lg shadow-lg flex-col ${style}`}>{children}</div>
        </div>
    );
};
