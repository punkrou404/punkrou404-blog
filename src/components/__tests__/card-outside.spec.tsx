import React from 'react';
import renderer from 'react-test-renderer';
import { CardOutside } from '../card-outside';

describe('CardOutside', () => {
    it('can action', () => {
        const component = renderer.create(<CardOutside>{`aaa`}</CardOutside>);
        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('cannot action', () => {
        const component = renderer.create(<CardOutside action={false}>{`aaa`}</CardOutside>);
        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });
});
