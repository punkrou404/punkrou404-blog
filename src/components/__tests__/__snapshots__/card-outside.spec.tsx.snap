// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardOutside can action 1`] = `
<div
  className="w-full flex-col p-3 text-gray-500"
>
  <div
    className="p-4 bg-white rounded-lg shadow-lg flex-col hover:bg-gray-200 active:bg-gray-400"
  >
    aaa
  </div>
</div>
`;

exports[`CardOutside cannot action 1`] = `
<div
  className="w-full flex-col p-3 text-gray-500"
>
  <div
    className="p-4 bg-white rounded-lg shadow-lg flex-col "
  >
    aaa
  </div>
</div>
`;
